src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,191 @@
+
+          if (onError) {
+            onError();
+          }
+        });
+    };
+
+    const handleChange: OnChangeFunc = (
+      todoId,
+      propName,
+      newPropValue,
+      onError,
+    ) => {
+      hideError();
+      setIsWaiting(true);
+
+      updateTodoOnServer(todoId, { [propName]: newPropValue })
+        .then(() => onChangeTodo(todoId, propName, newPropValue))
+        .catch(() => {
+          showError(ErrorType.Update);
+
+          if (onError) {
+            onError();
+          }
+        })
+        .finally(() => setIsWaiting(false));
+    };
+
+    const handleStatusChange = (): void => {
+      handleChange(id, 'completed', !completed);
+    };
+
+    const handleTitleChange = (): void => {
+      const newTitle = editedTitle.trim();
+
+      setIsEditAllowed(false);
+      setEditedTitle(newTitle);
+
+      if (newTitle === title) {
+        return;
+      }
+
+      const onError = (): void => {
+        setEditedTitle(title);
+      };
+
+      if (!newTitle) {
+        handleDeletingTodo(onError);
+
+        return;
+      }
+
+      handleChange(id, 'title', newTitle, onError);
+    };
+
+    const cancelTitleChange = (
+      event: React.KeyboardEvent<HTMLInputElement>,
+    ): void => {
+      if (event.key !== 'Escape') {
+        return;
+      }
+
+      setIsEditAllowed(false);
+      setEditedTitle(title);
+    };
+
+    return (
+      <div
+        className={classNames('todo', {
+          completed,
+        })}
+      >
+        <label className="todo__status-label">
+          <input
+            type="checkbox"
+            className="todo__status"
+            checked={completed}
+            onChange={handleStatusChange}
+          />
+        </label>
+
+        {isEditAllowed ? (
+          <form
+            onSubmit={(event) => {
+              event.preventDefault();
+
+              handleTitleChange();
+            }}
+          >
+            <input
+              type="text"
+              className="todo__title-field"
+              placeholder="Empty todo will be deleted"
+              value={editedTitle}
+              onChange={(event) => setEditedTitle(event.target.value)}
+              onBlur={handleTitleChange}
+              onKeyUp={cancelTitleChange}
+              ref={editFormRef}
+            />
+          </form>
+        ) : (
+          <>
+            <span
+              role="button"
+              tabIndex={0}
+              aria-label="Press Enter to edit the title"
+              className="todo__title"
+              onKeyUp={(event) => {
+                if (event.key !== 'Enter') {
+                  return;
+                }
+
+                setIsEditAllowed(true);
+              }}
+              onDoubleClick={() => setIsEditAllowed(true)}
+            >
+              {title}
+            </span>
+
+            <button
+              type="button"
+              className="todo__remove"
+              onClick={() => handleDeletingTodo()}
```

Comment:
You can shorten it
```suggestion
              onClick={handleDeletingTodo}
```

Reply:
But actually, I can't) There is a reason to write in such a way - handleDeletingTodo has an optional param onError, which describes an additional behaviour for handling an error.